homeassistant/components/intellifire/climate.py

```diff
@@ -0,0 +1,113 @@
+
+    async_add_entities(
+        IntellifireClimate(
+            coordinator=coordinator,
+            description=description,
+        )
+        for description in INTELLIFIRE_CLIMATES
+    )
+
+
+class IntellifireClimate(IntellifireEntity, ClimateEntity):
+    """Intellifire climate entity."""
+
+    entity_description: ClimateEntityDescription
+
+    _attr_hvac_modes = [HVAC_MODE_HEAT, HVAC_MODE_OFF]
+    _attr_min_temp = 0
+    _attr_max_temp = 37
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = TEMP_CELSIUS
+    last_temp = DEFAULT_THERMOSTAT_TEMP
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return current hvac mode."""
+        if self.coordinator.read_api.data.thermostat_on:
+            return HVAC_MODE_HEAT
+        return HVAC_MODE_OFF
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Turn on thermostat by setting a target temperature."""
+        raw_target_temp = kwargs[ATTR_TEMPERATURE]
+        self.last_temp = int(raw_target_temp)
+        LOGGER.debug(
+            "Setting target temp to %sc %sf",
+            int(raw_target_temp),
+            (raw_target_temp * 9 / 5) + 32,
+        )
+        await self.coordinator.control_api.set_thermostat_c(
+            fireplace=self.coordinator.control_api.default_fireplace,
+            temp_c=self.last_temp,
+        )
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return float(self.coordinator.read_api.data.temperature_c)
+
+    @property
+    def target_temperature(self) -> float:
+        """Return target temperature."""
+        return float(self.coordinator.read_api.data.thermostat_setpoint_c)
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set HVAC mode to normal or thermostat control."""
+        LOGGER.info(
+            "Setting mode to [%s] - using last temp: %s", hvac_mode, self.last_temp
+        )
+
+        if hvac_mode == HVAC_MODE_HEAT:
+            # 1) Set the desired target temp
+            await self.coordinator.control_api.set_thermostat_c(
+                fireplace=self.coordinator.control_api.default_fireplace,
+                temp_c=self.last_temp,
+            )
+
+            # 2) Make sure the fireplace is on!
+            if not self.coordinator.read_api.data.is_on:
+                await self.coordinator.control_api.flame_on(
+                    fireplace=self.coordinator.control_api.default_fireplace,
+                )
+
+        elif hvac_mode == HVAC_MODE_OFF:
+            await self.coordinator.control_api.turn_off_thermostat(
+                fireplace=self.coordinator.control_api.default_fireplace
+            )
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Enable entity if it exists."""
+        return bool(self.coordinator.data.has_thermostat)
```

Comment:
If it doesn't have a thermostat we should instead return early in `async_setup_entry` and not create the entity

Reply:
Should I nix this block then?

Reply:
yes